src/org/omegat/gui/editor/EditorUtils.java

```diff
@@ -99,6 +100,41 @@ public static int getWordEnd(JTextComponent c, int offs) throws BadLocationExcep
         return result;
     }
 
+    private static int getWordBoundary(JTextComponent c, int offs, boolean end) throws BadLocationException {
+        int result = offs;
+        Element line = Utilities.getParagraphElement(c, offs);
+        if (line == null) {
+            throw new BadLocationException("No word at " + offs, offs);
+        }
+        int lineStart = line.getStartOffset();
+        Document doc = c.getDocument();
+        int lineEnd = Math.min(line.getEndOffset(), doc.getLength());
+        if  (lineEnd - lineStart > 0) {
+            String lineString = doc.getText(lineStart, lineEnd - lineStart);
+            Locale locale = c.getLocale();
```

Comment:
It is better to leave a comment that explain an intention. 

```suggestion
            // Detection of target string locale.
            // A default is UI component locale, and when OmegaT project
            // is loaded, it uses a source or a target language as a
            // processing locale.
            Locale locale = c.getLocale();
```